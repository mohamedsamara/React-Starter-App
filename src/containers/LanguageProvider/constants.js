/*
 *
 * LanguageProvider constants
 *
 */

export const SET_LOCALE = 'src/LanguageProvider/SET_LOCALE';
