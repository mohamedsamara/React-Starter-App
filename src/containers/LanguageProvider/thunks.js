/*
 *
 * LanguageProvider thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import { handleSetLocale } from './actions';

export function setLocale(lang) {
  return function(dispatch, getState) {
    localStorage.appLang = lang;
    return dispatch(handleSetLocale(lang));
  };
}
