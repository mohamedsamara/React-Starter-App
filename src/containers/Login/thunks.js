/*
 * 
 * Login
 * 
 */


// Do this in every file where you use `fetch`
import fetch from 'cross-fetch'
import validator from "validator";
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';


import {} from './actions';


/*
export function fetch() {
  return function(dispatch, getState) {
    return fetch('', {
      method: 'GET'
    })
      .then(response => {
        return response.json();
      })
      .then(data => {
      })
      .catch(error => {});
  };
}
*/
