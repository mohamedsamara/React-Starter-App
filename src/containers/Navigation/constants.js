/*
 *
 * Navigation constants
 *
 */

export const TOGGLE_DRAWER = 'src/Navigation/TOGGLE_DRAWER';
