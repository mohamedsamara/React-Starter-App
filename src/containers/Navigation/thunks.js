/*
 *
 * Navigation
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import validator from 'validator';

import { handleToggleDrawer } from './actions';

// Toggle Drawer
export function toggleDrawer(side, open) {
  const isDrawerOpen = {};
  isDrawerOpen.side = side;
  isDrawerOpen.open = open;

  return function(dispatch, getState) {
    return dispatch(handleToggleDrawer(isDrawerOpen));
  };
}
