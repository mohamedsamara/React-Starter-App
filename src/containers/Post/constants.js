/*
 *
 * Post constants
 *
 */

export const FETCH_POSTS_REQUEST = 'src/Post/FETCH_POSTS_REQUEST';
export const FETCH_POSTS_SUCCESS = 'src/Post/FETCH_POSTS_SUCCESS';
export const HANLDE_POSTS_FAILURE = 'src/Post/HANLDE_POSTS_FAILURE';
export const DELETE_POST_SUCCESS = 'src/Post/DELETE_POST_SUCCESS';
