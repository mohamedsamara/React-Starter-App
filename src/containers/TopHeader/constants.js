/*
 *
 * TopHeader constants
 *
 */

export const OPEN_MENU = 'src/TopHeader/OPEN_MENU';
export const CLOSE_MENU = 'src/TopHeader/CLOSE_MENU';
