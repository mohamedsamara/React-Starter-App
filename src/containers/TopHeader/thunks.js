/*
 *
 * TopHeader thunks
 *
 */

// Do this in every file where you use `fetch`
import fetch from 'cross-fetch';
import { toastr } from 'react-redux-toastr';
import { push, goBack } from 'connected-react-router';

import { handleOpenMenu, handleCloseMenu } from './actions';

// Open Menu
export function openMenu(target) {
  return function(dispatch, getState) {
    return dispatch(handleOpenMenu(target));
  };
}

// Close Menu
export function closeMenu() {
  return function(dispatch, getState) {
    return dispatch(handleCloseMenu());
  };
}
