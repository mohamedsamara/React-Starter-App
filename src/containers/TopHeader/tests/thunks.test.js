import { openMenu, closeMenu } from '../thunks';
import { OPEN_MENU, CLOSE_MENU } from '../constants';

describe('topHeader Thunk', () => {
  it('should dispatch correct actions when handling openMenu', () => {
    const mockDispatch = jest.fn();

    const target =
      '<button class="MuiButtonBase-root-29 MuiIconButton-root-89 MuiIconButton-colorInherit-90" tabindex="0" type="button" aria-label="More" aria-haspopup="true" aria-owns="long-menu"><span class="MuiIconButton-label-94"><svg class="MuiSvgIcon-root-95" focusable="false" viewBox="0 0 24 24" aria-hidden="true" role="presentation"><path fill="none" d="M0 0h24v24H0z"></path><path d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"></path></svg></span><span class="MuiTouchRipple-root-142"></span></button>';

    const expected = {
      type: OPEN_MENU,
      target,
    };

    expect(typeof openMenu()).toEqual('function');
    openMenu(target)(mockDispatch);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
    expect(mockDispatch).toMatchSnapshot();
  });

  it('should dispatch correct actions when handling closeMenu', () => {
    const mockDispatch = jest.fn();

    const expected = {
      type: CLOSE_MENU,
    };

    expect(typeof openMenu()).toEqual('function');
    closeMenu()(mockDispatch);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
    expect(mockDispatch).toMatchSnapshot();
  });
});
