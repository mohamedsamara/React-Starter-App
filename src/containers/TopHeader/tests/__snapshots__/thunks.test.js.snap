// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`topHeader Thunk should dispatch correct actions when handling closeMenu 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "type": "src/TopHeader/CLOSE_MENU",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`topHeader Thunk should dispatch correct actions when handling openMenu 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "target": "<button class=\\"MuiButtonBase-root-29 MuiIconButton-root-89 MuiIconButton-colorInherit-90\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"More\\" aria-haspopup=\\"true\\" aria-owns=\\"long-menu\\"><span class=\\"MuiIconButton-label-94\\"><svg class=\\"MuiSvgIcon-root-95\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path fill=\\"none\\" d=\\"M0 0h24v24H0z\\"></path><path d=\\"M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z\\"></path></svg></span><span class=\\"MuiTouchRipple-root-142\\"></span></button>",
        "type": "src/TopHeader/OPEN_MENU",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
